indy_node/server/domain_req_handler.py

```diff
@@ -210,14 +212,33 @@ def _validateAttrib(self, req: Request):
                                        'attribute for it'.
                                        format(TARGET_NYM))
 
-        if op.get(TARGET_NYM) and op[TARGET_NYM] != req.identifier and \
-                not self.idrCache.getOwnerFor(op[TARGET_NYM],
-                                              isCommitted=False) == origin:
-            raise UnauthorizedClientRequest(
-                req.identifier,
-                req.reqId,
-                "Only identity owner/guardian can add attribute "
-                "for that identity")
+        is_owner = self.idrCache.getOwnerFor(op[TARGET_NYM],
```

Comment:
It looks like `TARGET_NYM` is an optional field, while in `prepare_attr_for_state` and validation we assume it may be not.
I suggest to remove `optional` tagfrom Atrrib's `TARGET_NYM` (see `ClientAttribOperation`)

Reply:
agreed